gnomad/assessment/sanity_checks.py

```diff
@@ -190,3 +212,744 @@ def compare_row_counts(ht1: hl.Table, ht2: hl.Table) -> bool:
+                    check_field_right: t_xnonpar.info[check_field_right],
+                }
+            ),
+        }
+
+    generic_field_check_loop(t_xnonpar, field_check_expr, verbose)
+
+
+def missingness_sanity_checks(
+    t: Union[hl.MatrixTable, hl.Table],
+    info_metrics: List[str],
+    non_info_metrics: List[str],
+    n_sites: int,
+    missingness_threshold: float,
+) -> None:
+    """
+    Check amount of missingness in all row annotations.
+
+    Print metric to sdout if the metric annotations missingness exceeds the missingness_threshold.
+
+    :param t: Input MatrixTable or Table.
+    :param info_metrics: List of metrics in info struct of input Table.
+    :param non_info_metrics: List of row annotations minus info struct from input Table.
+    :param n_sites: Number of sites in input Table.
+    :param missingness_threshold: Upper cutoff for allowed amount of missingness.
+    :return: None
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    logger.info(
+        "Missingness threshold (upper cutoff for what is allowed for missingness checks): %.2f",
+        missingness_threshold,
+    )
+    metrics_missing = {}
+    for x in info_metrics:
+        metrics_missing[x] = hl.agg.sum(hl.is_missing(t.info[x]))
+    for x in non_info_metrics:
+        metrics_missing[x] = hl.agg.sum(hl.is_missing(t[x]))
+    output = dict(t.aggregate(hl.struct(**metrics_missing)))
+
+    n_fail = 0
+    for metric, n_missing in output.items():
+        if n_missing / n_sites > missingness_threshold:
+            logger.info(
+                "FAILED missingness check for %s: %d sites or %.2f%% missing",
+                metric,
+                n_missing,
+                (100 * n_missing / n_sites),
+            )
+            n_fail += 1
+        else:
+            logger.info(
+                "Passed missingness check for %s: %d sites or %.2f%% missing",
+                metric,
+                n_missing,
+                (100 * n_missing / n_sites),
+            )
+    logger.info("%d missing metrics checks failed", n_fail)
+
+
+def vcf_field_check(
+    t: Union[hl.MatrixTable, hl.Table],
+    header_dict: Dict[str, Dict[str, Dict[str, str]]],
+    row_annotations: List[str],
+    hists: List[str] = HISTS,
+) -> bool:
+    """
+    Check that all VCF fields and descriptions are present in input Table and VCF header dictionary.
+
+    :param t: Input MatrixTable or Tableto be exported to VCF.
+    :param header_dict: VCF header dictionary.
+    :param row_annotations: List of row annotations in MatrixTable.
+    :param hists: List of variant histogram annotations. Default is HISTS.
+    :return: Boolean with whether all expected fields and descriptions are present.
+    """
+    t = t.rows() if isinstance(t, hl.MatrixTable) else t
+
+    # Confirm all VCF fields/descriptions are present before exporting
+    hist_fields = []
+    for hist in hists:
+        hist_fields.extend(
```

Comment:
this was written before the extra hists were dropped from the release file. maybe we should stop adding the extra hists here to `hist_fields` and then update the check below (line 838)?